client/signup/jetpack-connect/site-card.jsx

```diff
@@ -47,7 +48,7 @@ class SiteCard extends Component {
 
 		return (
 			<CompactCard className="jetpack-connect__site">
-				<QuerySites allSites />
+				<QuerySites siteId={ parseInt( client_id ) } />
```

Comment:
Is there a case where `client_id` can be returned from the API as a non-numeric value? That could query site with ID `0` or `1`, so let's confirm that this is not a possible case.

That also applies for when we perform the network request in the `authorize()` action.